src/main/networking/vtunnel.ts

```diff
@@ -31,24 +44,66 @@ export class vtunnel {
     },
   });
 
-  async generateVtunnelConfig() {
+  private async generateConfig() {
     const conf = {
-      tunnel: [
-        {
-          'handshake-port':          credServerHandshakePort,
-          'vsock-host-port':         credServerVsockPort,
-          'peer-address':            localHost,
-          'peer-port':               credServerPeerPort,
-          'upstream-server-address': `${ localHost }:${ credServerPort }`,
-        }
-      ]
+      tunnel: this._vtunnelConfig.map(c => ({
+        'handshake-port':          c.handshakePort,
+        'vsock-host-port':         c.vsockHostPort,
+        'peer-address':            c.peerAddress,
+        'peer-port':               c.peerPort,
+        'upstream-server-address': c.upstreamServerAddress,
+      }))
     };
+
     const configYaml = yaml.stringify(conf);
 
     await fs.promises.writeFile(getVtunnelConfigPath(), configYaml, 'utf8');
   }
+
+  /**
+   * addTunnel adds a new configuration to an existing list of configs.
+   */
+  addTunnel(config: VtunnelConfig) {
+    this._vtunnelConfig.push(config);
```

Comment:
(We may decide in the future to dedupe based on name, but that's not a thing yet.)